core/base-service/route.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const Joi = require('joi')
+const pathToRegexp = require('path-to-regexp')
+
+function makeFullUrl(base, partialUrl) {
+  return `/${[base, partialUrl].filter(Boolean).join('/')}`
+}
+
+const routeSchema = Joi.object({
+  base: Joi.string().allow(''),
+  pattern: Joi.string().allow(''),
+  format: Joi.string(),
+  capture: Joi.alternatives().when('format', {
+    is: Joi.string().required(),
+    then: Joi.array().items(Joi.string().required()),
+  }),
+  queryParams: Joi.array().items(Joi.string().required()),
+})
+  .xor('pattern', 'format')
+  .required()
+
+function assertValidRoute(route, message = undefined) {
+  Joi.assert(route, routeSchema, message)
+}
+
+function prepareRoute({ base, pattern, format, capture }) {
+  let regex, captureNames
+  if (pattern === undefined) {
+    regex = new RegExp(
+      `^${makeFullUrl(base, format)}\\.(svg|png|gif|jpg|json)$`
+    )
+    captureNames = capture || []
```

Comment:
Should the schema validation ensure `capture` is set by the time we get here?

Reply:
There’s no `.required()` on the array, so I think the answer is no. However if it’s easy to make that happen I’d be down to make the change.

Reply:
Yeah this made more sense to me as I went through the rest of the code. Even if there were some Joi magic there's no guaranteed it would have been called. 

I get it now 👍